routes/projects/files/create-update.js

```diff
@@ -0,0 +1,28 @@
+/*
+ * PUT /projects/:username/:projectName/files/:pathToFile
+ * The :pathToFile parameter must be URI encoded
+ * Create or replace the contents for the given file path
+ * The message body that would normally be expected by this route
+ * will look something like:
+ * {
+ *   type: <string> content-type required,
+ *   path: <string> required,
+ *   size: <integer> required,
```

Comment:
This isn't necessary, since it can be determined by the size of the body contents.  I'd remove it.